.github/workflows/ci.yml

```diff
@@ -0,0 +1,33 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:  # Support latest and one minor back
+        go: ['1.15', '1.16']
+    env:
+      GOFLAGS: -mod=readonly
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          stable: 'false'
+          go-version: ${{ matrix.go }}
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Run tests
+        env:
+          SKIP_NET_TESTS: "true"
```

Comment:
If you want to not run some tests there just specify the files which you want to include/exclude. I would like to not  have if (environment variable) in test files.

Reply:
I don't think you can specify files with `go test`. Only packages. AFAIK this is an accepted practice in Go to skip tests on certain platforms. The alternative is always skipping them, because currently they cannot be executed, but I think that's worse.

Another alternative is using `go test -run` and excluding the names of tests, but there are quite a few tests to exclude, so I don't think that's better either.

What would you like me to do?

Reply:
Ideally, we would like to make this workflow run all tests. It can even start dockers with eureka running in them and test net tests as well. 

If that's too much work for you, I would prefer specifying which test you want to run with regexp Somehow useful: https://stackoverflow.com/questions/16935965/how-to-run-test-cases-in-a-specified-file

I'm perfectly ok with you changing the name of the tests to your liking. I'm just not a fan of using environment variables for the tests flows. Environment variables are not well supported by all IDEs and multi-threaded execution.

Reply:
I agree that ideally all tests should be run, especially in CI, but I didn't want to spend time with figuring out why the tests don't pass. This was never meant to be permanent.

Let me try to come up with something.

Reply:
Replaced env vars with Go's short test feature.

Reply:
Normally I would be ok with not fighting with already failing tests, but in this scenario when we upgraded Go version we don't know if we didn't introduce new errors and I want all test which is there to pass. It's somehow a problem of a repository that is not actively maintained :(

Reply:
I actually got the tests passing with a few exceptions:

- IMHO mandating IP address for containers is a bad approach and it simply doesn't work in CI. I added a docker-compose based test environment, but the tests validating IP addresses fail. I'm not sure how to solve that yet. Personally, I'd just leave that check out. I don't think it makes any sense TBH.
- Some tests seem to be flaky. I don't know yet if it's because Eureka starts up slowly or something else.

I don't think the Go update broke any of those tests. For one, I wasn't able to run them before the upgrade. Relying on a static environment is not a good practice in my opinion (using env vars is one way to feed parameters into tests) and I think it has something to do with the breaking tests.